src/sagemaker_tensorflow_container/training.py

```diff
@@ -0,0 +1,212 @@
+        _run_ps(env, tf_config['cluster'])
+        logger.info('Launching worker process')
+        _run_worker(env, cmd_args, tf_config)
+
+        if not _is_host_master(env.hosts, env.current_host):
+            _wait_until_master_is_down(env.hosts[0])
+
+    else:
+
+        mpi_enabled = env.additional_framework_parameters.get('sagemaker_mpi_enabled')
+
+        if mpi_enabled:
+            runner_type = framework.runner.MPIRunnerType
+        else:
+            runner_type = framework.runner.ProcessRunnerType
+
+        framework.entry_point.run(env.module_dir, env.user_entry_point, cmd_args, env.to_env_vars(),
+                                  runner=runner_type)
+
+
+def _log_model_missing_warning(model_dir):
+    pb_file_exists = False
+    file_exists = False
+    for dirpath, dirnames, filenames in os.walk(model_dir):
+        if filenames:
+            file_exists = True
+        for f in filenames:
+            if 'saved_model.pb' in f or 'saved_model.pbtxt' in f:
+                pb_file_exists = True
+                path, direct_parent_dir = os.path.split(dirpath)
+                if not str.isdigit(direct_parent_dir):
+                    logger.warn('Your model will NOT be servable with SageMaker TensorFlow Serving containers.'
+                                'The SavedModel bundle is under directory \"{}\", not a numeric name.'
+                                .format(direct_parent_dir))
+
+    if not file_exists:
+        logger.warn('No model artifact is saved under path {}.'
+                    ' Your training job will not save any model files to S3.\n'
+                    'For details of how to construct your training script see:\n'
+                    'https://github.com/aws/sagemaker-python-sdk/tree/master/src/sagemaker/tensorflow#adapting-your-local-tensorflow-script' # noqa
+                    .format(model_dir))
+    elif not pb_file_exists:
+        logger.warn('Your model will NOT be servable with SageMaker TensorFlow Serving container.'
+                    'The model artifact was not saved in the TensorFlow SavedModel directory structure:\n'
+                    'https://www.tensorflow.org/guide/saved_model#structure_of_a_savedmodel_directory')
+
+
+def _model_dir_with_training_job(model_dir, job_name):
+    if model_dir.startswith('/opt/ml'):
+        return model_dir
+    else:
+        return '{}/{}/model'.format(model_dir, job_name)
+
+
+def main():
+    """Training entry point
+    """
+    hyperparameters = framework.env.read_hyperparameters()
+    env = framework.training_env(hyperparameters=hyperparameters)
+
+    user_hyperparameters = env.hyperparameters
+
+    # If the training job is part of the multiple training jobs for tuning, we need to append the training job name to
+    # model_dir in case they read from/write to the same object
+    if '_tuning_objective_metric' in hyperparameters:
+        model_dir = _model_dir_with_training_job(hyperparameters.get('model_dir'), env.job_name)
```

Comment:
it seems that `hyperparameters` does not have 'model_dir' while running my tuning job. Should `hyperparameters` here be `user_hyperparameters` instead? @mvsusp